File: golang/cmd/domains/domains.go
// Usage: TOKEN=token go run cmd/domains/domains.go
package main

import (
	"context"
	"fmt"
	"os"
	"strconv"

	"github.com/dnsimple/dnsimple-go/v5/dnsimple"
	"golang.org/x/oauth2"
)

func main() {
	oauthToken := os.Getenv("TOKEN")
	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: oauthToken})
	tc := oauth2.NewClient(context.Background(), ts)

	client := dnsimple.NewClient(tc)
	client.BaseURL = "https://api.sandbox.dnsimple.com"

	// get the current authenticated account (if you don't know who you are)
	whoamiResponse, err := client.Identity.Whoami(context.Background())
	if err != nil {
		fmt.Printf("Whoami() returned error: %v\n", err)
		os.Exit(1)
	}

	accountId := strconv.FormatInt(whoamiResponse.Data.Account.ID, 10)

	domainsResponse, err := client.Domains.ListDomains(context.Background(), accountId, nil)
	if err != nil {
		fmt.Printf("ListDomains() returned error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("%+v\n", domainsResponse)
}

File: csharp/Examples/Domains.cs
using dnsimple;

namespace DnsimpleExamples.Examples;

// Usage: TOKEN=your-token dotnet run --framework net10.0 -- domains
public static class Domains
{
    public static void Run(string[] args)
    {
        var token = Environment.GetEnvironmentVariable("TOKEN");
        if (string.IsNullOrEmpty(token))
        {
            Console.Error.WriteLine("The TOKEN environment variable is required");
            Environment.Exit(1);
        }

        var client = new Client();
        client.ChangeBaseUrlTo("https://api.sandbox.dnsimple.com");
        client.AddCredentials(new OAuth2Credentials(token));

        // get the current authenticated account (if you don't know who you are)
        var accountId = client.Identity.Whoami().Data.Account.Id;

        // list the domains in the account
        try
        {
            var domains = client.Domains.ListDomains(accountId).Data;
            foreach (var domain in domains)
            {
                Console.WriteLine($"Domain: Id={domain.Id}, Name={domain.Name}, State={domain.State}, ExpiresAt={domain.ExpiresAt}");
            }
        }
        catch (DnsimpleException e)
        {
            Console.Error.WriteLine($"ListDomains() returned error: {e.Message}");
            Environment.Exit(1);
        }
    }
}
